### PR TITLE
Added blast.py script

### PR DESCRIPTION
**Pull request by aays**
Adds `blast.py`, a general purpose blast script that converts input to a BLAST+ command line call. Allows for blasting a file against either an NCBI database or an existing file. This addresses issue #189.

### CODE REVIEW

File: blast.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+cline.extend(['-query', args.query])
+cline.extend(['-evalue', str(args.evalue)])
+cline.extend(['-out', str(args.output)])
+
+# subject/db parameter
+subject = os.path.splitext(args.subject)
+if subject[1] == '':  # db name provided
+    cline.extend(['-db', subject[0]])
+elif subject[1] == '.fasta' or subject[1] == '.fa':
```
Comment (QuLogic) [inline blast.py]: `subject[1] in ('.fa', '.fasta')`

File: blast.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    cline.extend(['-subject', ''.join(subject)])
```
Comment (QuLogic) [inline blast.py]: Isn't `''.join(subject)` the same as `args.subject`?

File: blast.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    print('Please specify either a db name or an input .fasta file.')
+
+# optional parameters
+cline.extend(['-num_alignments', str(args.num_alignments)])
+cline.extend(['-reward', str(args.reward)])
+cline.extend(['-penalty', str(args.penalty)])
+
+print('Command line BLAST+')
+print('Options entered:\n')
+print(' '.join(cline), '\n')
```
Comment (QuLogic) [inline blast.py]: You know, people ask why print should be a function, and you don't have to change this, but here's what you can do with a function:
```
print(*cline, eol='\n\n')
```

File: blast.py (hunk 2 above)
Comment (aays) [inline blast.py]: .........yes

File: blast.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+# subject/db parameter
+subject = os.path.splitext(args.subject)
+if subject[1] == '':  # db name provided
+    cline.extend(['-db', subject[0]])
+elif subject[1] in ('.fasta', '.fa'):
+    cline.extend(['-subject', args.subject])
+else:
+    sys.exit()
+    print('Error in db/subject specified.')
+    print('Please specify either a db name or an input .fasta file.')
```
Comment (mbonsma) [inline blast.py]: I guess put `sys.exit()` at the end here?

**Pull request merged**